This should correct #5

esabelhaus opened the pull request:
I ran into the same blocker as mentioned in #5, decided to fix it up for ya.

I added a case statement to include elrepo, and updated packages logic to differentiate between ubuntu/debian and the rest

tas50 commented on recipes/default.rb:
```diff
@@ -20,8 +20,22 @@
 
-package 'drbd8-utils' do
-  action :install
+case node['platform']
```
this should be platform_family which will include opensuse, opensuseleap, centos, scientific, oracle, and amazon

esabelhaus replied:
Going by the support from DRBD, this should only support RHEL, Fedora, Suse, Debian, and Ubuntu.

I have only tested this on Ubuntu 14.04 and CentOS 6.5, but it should work on all centos 6.x 

http://www.drbd.org/en/doc/users-guide-83/ch-install-packages

Perhaps I should modify the Metadata file to reference only supported OSes?

-Haus

tas50 commented on metadata.rb:
```diff
@@ -6,6 +6,8 @@
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
 version '0.10.0'
 
+depends 'yum-elrepo'
```
This will need to get called out in the readme under the requirements section.  See https://github.com/chef-cookbooks/activemq/blob/master/README.md#cookbooks

esabelhaus commented:
Would you like me to sort out the above comments?

-Haus

esabelhaus commented on recipes/default.rb:
```diff
@@ -20,8 +20,22 @@
+end
+
+drbd_packages = value_for_platform_family(
+  ['rhel', 'fedora', 'suse'] => ['kmod-drbd84', 'drbd84-utils'],
```
Specifically, updating this block

cheeseplus commented:
Could this be rebased? If not I may merge it and resolve the conflict myself but wanted to give a chance to do it the right way first!

esabelhaus commented:
I'll give it a swing, should have some time today :smiley:

esabelhaus commented:
Converging the centos-6.7 suite locally now

esabelhaus commented:
got a few minor tweaks, will be pushed shortly

esabelhaus commented:
@tas50 any objection to merging this? 
I've tested converging with both 6.7 and 6.8 without fail.
I've added chefspec to ensure converge is successful.
I've got everything happy with FC and RC.
let me know if I'm missing anything :smiley:

tas50 commented:
@esabelhaus You'll need to revert the changes to the readme that reverted the formatting fixes. Also you only want to include the yum-elrepo recipe if you're on the RHEL platform family. Otherwise we're trying to load that on Ubuntu/Debian as well.

tas50 commented on .kitchen.yml:
```diff
@@ -14,8 +14,14 @@ platforms:
   - name: ubuntu-16.04
     run_list:
     - recipe[apt::default]
+  - name: centos-6.7
```
You can remove 6.7 here. We just test on the latest image we have of each major release.

tas50 commented on README.md:
```diff
@@ -1,60 +1,57 @@
-# drbd Cookbook
+drbd Cookbook
```
The format changes and removal of the requirements section needs to get reverted.

tas50 commented on spec/unit/recipes/default_spec.rb:
```diff
@@ -11,3 +11,15 @@
+describe 'default recipe on Ubuntu 14.04' do
+  let(:chef_run) do
+    ChefSpec::ServerRunner.new do |node|
+      node.automatic[:platform_family] = 'rhel'
```
Check out the Chefspec page for the proper way to mock out a CentOS node completely. This will technically work, but it should be done the proper way.

esabelhaus replied:
Going to update your existing spec and add another for ubuntu 16.04 based off this example https://github.com/sethvargo/chefspec/blob/master/examples/server/spec/node_spec.rb

esabelhaus commented:
@tas50 I've modified the unit tests to follow the chefspec ServerRunner.new() format, let me know if there's anything else you'd like me to tweak :smiley: 

-Haus

esabelhaus commented:
@tas50 any movement on this?

esabelhaus commented:
Just rebased to master, if something is busted, I will fix it later today. @tas50 I have modified the code which you requested previously, is there any reason this can't be merged?

tas50 commented:
@esabelhaus I've made some changes on master for a few things that weren't necessary in the change. Thanks for putting it together and getting this pushed through.
